app/controllers/admin/api/application_plan_limits_controller.rb

```diff
@@ -14,15 +14,19 @@ class Admin::Api::ApplicationPlanLimitsController < Admin::Api::BaseController
   ##~ op.description = "Returns the list of all limits associated to an application plan."
   ##~ op.group = "application_plan_limits"
   #
+  ##~ @parameter_per_page_50 = {:name => "per_page", :description => "Number of results per page. Default and max is 50.", :dataType => "int", :paramType => "query", :defaultValue => "50"}
+  #
   ##~ op.parameters.add @parameter_access_token
+  ##~ op.parameters.add @parameter_page
+  ##~ op.parameters.add @parameter_per_page_50
   ##~ op.parameters.add @parameter_application_plan_id_by_id_name
   #
-
   def index
-    respond_with(usage_limits)
+    respond_with(usage_limits.paginate(page: current_page, per_page: per_page))
```

Comment:
I think a benchmark for this could be good to do. It is performance issue anyway.

Reply:
yep, benchmark sounds good.

Reply:
```
50 usage limits
Admin::Api::ApplicationPlanLimitsControllerTest
Measure Mode: wall_time
Thread ID: 70283998525840
Fiber ID: 70284006297440
Total: 1.915027
```

```
160 usage limits
Measure Mode: wall_time
Thread ID: 70283998525840
Fiber ID: 70284006297440
Total: 5.448363
Sort by: self_time
```

Reply:
Do you know what is making it slow?
N+1 requests? Missing index or rendering?

Reply:
lots of queriers, i'm on it.

Reply:
it's a bit better.. 

```
50 usage limits
Measure Mode: wall_time
Thread ID: 70289929263520
Fiber ID: 70289961327280
Total: 1.101826
Sort by: self_time
```